core/src/main/scala/quasar/plugin/postgres/PostgresDestinationModule.scala

```diff
@@ -16,5 +16,132 @@
 
 package quasar.plugin.postgres
 
-object PostgresDestinationModule {
+import slamdata.Predef._
+
+import argonaut._, Argonaut._
+
+import cats.data.EitherT
+import cats.effect._
+import cats.implicits._
+
+import doobie._
+import doobie.hikari.HikariTransactor
+import doobie.implicits._
+
+import eu.timepit.refined.auto._
+
+import java.util.concurrent.Executors
+
+import quasar.api.destination.{DestinationError => DE, _}
+import quasar.concurrent.{BlockingContext, NamedDaemonThreadFactory}
+import quasar.connector.{DestinationModule, MonadResourceErr}
+
+import scala.concurrent.ExecutionContext
+import scala.util.control.NonFatal
+
+import scalaz.syntax.tag._
+
+object PostgresDestinationModule extends DestinationModule {
+
+  type InitErr = DE.InitializationError[Json]
+
+  // The duration, in seconds, to await validation of the initial connection.
+  val ValidationTimeoutSeconds: Int = 10
+
+  // Maximum number of database connections per-destination.
+  val ConnectionPoolSize: Int = 10
+
+  val destinationType: DestinationType = DestinationType("postgres", 1L)
+
+  def sanitizeDestinationConfig(config: Json): Json =
+    config.as[Config]
+      .map(_.sanitized.asJson)
+      .getOr(jEmptyObject)
+
+  def destination[F[_]: ConcurrentEffect: ContextShift: MonadResourceErr: Timer](
+      config: Json)
+      : Resource[F, Either[InitErr, Destination[F]]] = {
+
+    val cfg0: Either[InitErr, Config] =
+      config.as[Config].fold(
+        (err, c) =>
+          Left(DE.malformedConfiguration[Json, InitErr](
+            destinationType,
+            config,
+            err)),
+
+        Right(_))
+
+    val validateConnection: ConnectionIO[Either[InitErr, Unit]] =
+      FC.isValid(ValidationTimeoutSeconds) map { v =>
+        if (!v) Left(connectionInvalid(config)) else Right(())
+      }
+
+    val init = for {
+      cfg <- EitherT(cfg0.pure[Resource[F, ?]])
+
+      suffix <- EitherT.right(Resource.liftF(randomAlphaNum[F](6)))
```

Comment:
We should log this suffix. Thread names don't actually have to be unique (to my knowledge); their only purpose is just to make debugging easier, but obviously if we don't know what suffix to be expecting for a given destination, we might not know what threads to look for.

Reply:
Yeah, the suffix was only for debugging, will log it 👍